### PR TITLE
Add support for additional Toolbar items to MeasureToolsUiItemsProvider

### PR DESCRIPTION
**Pull request by saaaaaally**
Added addAdditionalToolbarItems to MeasureToolsUiItemsProvider so that applications can pass in additional measure tools to be displayed at the end of the measure tools list.

### CODE REVIEW

**Review by wilmaier — APPROVED** [review]

File: packages/itwin/measure-tools/src/ui-2.0/MeasureToolsUiProvider.tsx [hunk 1]
```diff
@@ -51,6 +52,10 @@ export class MeasureToolsUiItemsProvider implements UiItemsProvider {
     };
   }
 
+  public addAdditionalToolbarItems(toolbarItems: ToolItemDef[]): void {
```
Comment (aruniverse) [inline packages/itwin/measure-tools/src/ui-2.0/MeasureToolsUiProvider.tsx:55]: why wouldnt you just add additional tool bar items to MeasureToolsUiProviderOptions?

Comment (bsy-nicholasw) [reply packages/itwin/measure-tools/src/ui-2.0/MeasureToolsUiProvider.tsx:55]: Yeah, I'd put it as an option to the ctor. "Add" also here is really a replace, which I think consumers would not expect they'd be overwriting someone else's addition.

Comment (saaaaaally) [reply packages/itwin/measure-tools/src/ui-2.0/MeasureToolsUiProvider.tsx]: Moved additionalToolbarItems to MeasureToolsUiProviderOptions

**Review by bsy-nicholasw — APPROVED** [review]

File: packages/itwin/measure-tools/src/ui-2.0/MeasureToolsUiProvider.tsx [hunk 2]
```diff
@@ -32,11 +32,13 @@ export interface MeasureToolsUiProviderOptions {
   // If we check for sheet to 3d transformation when measuring in sheets
   enableSheetMeasurement?: boolean;
   stageUsageList?: string[];
+  additionalToolbarItems?: ToolItemDef[];
 }
 
 export class MeasureToolsUiItemsProvider implements UiItemsProvider {
   public readonly id = "MeasureToolsUiItemsProvider";
   private _props: RecursiveRequired<MeasureToolsUiProviderOptions>;
+  private _additionalToolbarItems: ToolItemDef[] = [];
```
Comment (aruniverse) [inline packages/itwin/measure-tools/src/ui-2.0/MeasureToolsUiProvider.tsx:41]: do we need this? we're gonna have 2 "places of truth"
_props would have additionalToolbarItems itself, but you set that to always be an empty array

Comment (saaaaaally) [reply packages/itwin/measure-tools/src/ui-2.0/MeasureToolsUiProvider.tsx]: Moved additionalToolbarItems into _props only

File: packages/itwin/measure-tools/src/ui-2.0/MeasureToolsUiProvider.tsx [hunk 3]
```diff
@@ -32,11 +32,12 @@ export interface MeasureToolsUiProviderOptions {
   // If we check for sheet to 3d transformation when measuring in sheets
   enableSheetMeasurement?: boolean;
   stageUsageList?: string[];
+  additionalToolbarItems?: ToolItemDef[];
 }
 
 export class MeasureToolsUiItemsProvider implements UiItemsProvider {
   public readonly id = "MeasureToolsUiItemsProvider";
-  private _props: RecursiveRequired<MeasureToolsUiProviderOptions>;
+  private _props: Omit<RecursiveRequired<MeasureToolsUiProviderOptions>, 'additionalToolbarItems'> & { additionalToolbarItems?: ToolItemDef[] };
```
Comment (aruniverse) [inline packages/itwin/measure-tools/src/ui-2.0/MeasureToolsUiProvider.tsx:40]: i was thinking `<RecursiveRequired<Omit<T, x>,> & { x: X },
but this should work too

**Review by aruniverse — APPROVED** [review]

**Pull request merged**